go/pkg/cli/checkout.go

```diff
@@ -36,92 +37,101 @@ func newCheckoutCommand() *cobra.Command {
-		if err != nil {
-			return err
-		}
-	}
-
+// Returns the repository requested by opts.repositoryURL
+func getRepositoryFromOpts(opts checkoutOpts) (repository.Repository, error) {
 	repositoryURL, projectDir, err := getRepositoryURLFromStringOrConfig(opts.repositoryURL)
 	if err != nil {
-		return err
+		return nil, err
 	}
+
 	repo, err := getRepository(repositoryURL, projectDir)
 	if err != nil {
-		return err
+		return nil, err
 	}
 
-	exists, err := files.FileExists(outputDir)
-	if err != nil {
-		return err
-	}
-	if exists {
-		isDir, err := files.IsDir(outputDir)
-		if err != nil {
-			return err
-		}
-		if !isDir {
-			return fmt.Errorf("Checkout path %q is not a directory", outputDir)
-		}
-	} else {
-		if err := os.MkdirAll(outputDir, 0755); err != nil {
-			return fmt.Errorf("Failed to create directory %q: %w", outputDir, err)
-		}
-	}
+	return repo, nil
+}
 
+// Returns the experiment and the most appropriate checkpoint for that experiment.
+func getExperimentAndCheckpoint(prefix string, repo repository.Repository) (*project.Experiment, *project.Checkpoint, error) {
 	proj := project.NewProject(repo)
 	result, err := proj.CheckpointOrExperimentFromPrefix(prefix)
 	if err != nil {
-		return err
+		return nil, nil, err
 	}
-
 	experiment := result.Experiment
 	checkpoint := result.Checkpoint
 
 	if checkpoint != nil {
 		console.Info("Checking out files from checkpoint %s and its experiment %s", checkpoint.ShortID(), experiment.ShortID())
-	} else {
-		// When checking out experiment, also check out best/latest checkpoint
-		checkpoint = experiment.BestCheckpoint()
-		if checkpoint != nil {
-			console.Info("Checking out files from experiment %s and its best checkpoint %s", experiment.ShortID(), checkpoint.ShortID())
-		} else {
-			checkpoint = experiment.LatestCheckpoint()
-			if checkpoint != nil {
-				console.Info("Checking out files from experiment %s and its latest checkpoint %s", experiment.ShortID(), checkpoint.ShortID())
-			} else {
-				console.Info("Checking out files from experiment %s", experiment.ShortID())
-			}
+		return experiment, checkpoint, nil
+	}
+
+	// When checking out experiment, also check out best/latest checkpoint
+	checkpoint = experiment.BestCheckpoint()
+	if checkpoint != nil {
+		console.Info("Checking out files from experiment %s and its best checkpoint %s", experiment.ShortID(), checkpoint.ShortID())
+		return experiment, checkpoint, nil
+	}
+
+	checkpoint = experiment.LatestCheckpoint()
+	if checkpoint != nil {
+		console.Info("Checking out files from experiment %s and its latest checkpoint %s", experiment.ShortID(), checkpoint.ShortID())
+		return experiment, checkpoint, nil
+	}
+
+	console.Info("Checking out files from experiment %s", experiment.ShortID())
+	return experiment, checkpoint, nil
+}
+
+// Handle errors related to the outputDir
+func validateOutputDir(outputDir string) error {
+	// FIXME(vastolorde95): If outputPath does not exist, there is no way to distinguish if
```

Comment:
I think it's okay for `--output-directory` to always be a directory. It would be nice to support the case you outline, but perhaps not worth the additional complexity. For now at least.

Reply:
Ack, deleted the FIXME